articles/introduction-to-scikit-learn-in-python/index.md

```diff
@@ -0,0 +1,297 @@
+
+```py
+from sklearn.preprocessing import LabelEncoder
+
+bands = ["Pink Floyd","Led Zeppelin","Pink Floyd","Foo Fighters","Queen","Queen","Pink Floyd","AC/DC","Foo Fighters","Led Zeppelin","Queen",
+           "Nirvana","AC/DC","The Doors","Queen","Fleetwood Mac","Nirvana"]
+
+# Invoking an instance of Label Encoder
+label_encoding = LabelEncoder()
+
+# Fit the labels
+encoded = label_encoding.fit(bands)
+
+print(encoded.transform(bands))
+
+# Output - [5 3 5 2 6 6 5 0 2 3 6 4 0 7 6 1 4]
+```
+If one were to look at the output, they would understand that the feature has been encoded. But mere numbers do not make any sense. Luckily, classes_ help us interpret what these labels are.
+
+```py
+#Iterate through the classes_ list and print them
+band_list = encoded.classes_
+
+for band_number in range(1,len(band_list)+1):
+    print(band_number, band_list[band_number-1])
+
+# Output
+# 1 AC/DC
+# 2 Fleetwood Mac
+# 3 Foo Fighters
+# 4 Led Zeppelin
+# 5 Nirvana
+# 6 Pink Floyd
+# 7 Queen
+# 8 The Doors
+```
+Note that the labels have been encoded in ascending order.
+
+If the "band_list" feature is one-hot encoded, it would be represented in 1's and 0's instead of decimals.
+
+```py
+import numpy as np
+from sklearn.preprocessing import OneHotEncoder
+
+band_list = np.array(["AC/DC","Fleetwood Mac","Foo Fighters","Led Zeppelin","Nirvana","Pink Floyd","Queen","The Doors"]).reshape(-1,1)
+
+# Invoking an instance of Label Encoder
+label_encoding = OneHotEncoder()
+
+# Fit the labels
+encoded = label_encoding.fit(band_list)
+
+print(encoded.transform(band_list).toarray())
+
+# Output
+# [[1. 0. 0. 0. 0. 0. 0. 0.]
+#  [0. 1. 0. 0. 0. 0. 0. 0.]
+#  [0. 0. 1. 0. 0. 0. 0. 0.]
+#  [0. 0. 0. 1. 0. 0. 0. 0.]
+#  [0. 0. 0. 0. 1. 0. 0. 0.]
+#  [0. 0. 0. 0. 0. 1. 0. 0.]
+#  [0. 0. 0. 0. 0. 0. 1. 0.]
+#  [0. 0. 0. 0. 0. 0. 0. 1.]]
+```
+
+### Scikit-Learn for Filling Missing Values
+
+Almost 70% of time and resources are spent on collecting and cleaning the dataset for every project. And when one deals with a real-life dataset, there are always missing values. Cleaning the dataset and handling missing data is important as many machine learning algorithms do not accommodate a missing attribute in the data. This is where Scikit-Learn's impute module comes into play. A simple way to deal with missing values is to remove the row of data with a missing value, which would mean losing valuable-yet-incomplete data. A better way is to replace the missing values with values that can be inferred from known data. One way would be to replace the missing data with the mean of that column.
+
+Missing values are encoded with NumPy's NaN (numpy.nan)
+
+The following are the temperatures recorded in Bloomington (in Fahrenheits) in Illinois in the month of February: [33.2,32.8,32.9,33.0,nan,33.2,33.4,33.1,32.6,32.5,32.5,33.1,33.0,nan,32.7,32.7,32.6,nan,32.6,32.9,32.8,32.8,32.5,32.6,nan,32.6,32.7,32.7,33.5, 33.7,33.9].
```

Comment:
Format numerical data, as stated earlier.

Reply:
Fixed it in the recent commit.